runtime/src/accounts_db.rs

```diff
@@ -773,15 +780,29 @@ impl AccountsDB {
 
         self.report_store_stats();
 
+        let mut key_create = Measure::start("key_create");
+        let mut pubkeys: Vec<Pubkey> = Vec::new();
+        {
+            let mut uncleaned_keys = self.uncleaned_keys.lock().unwrap();
+            let max_slot = max_clean_root.unwrap_or(u64::MAX);
+            let slots: Vec<Slot> = uncleaned_keys.keys().cloned().collect();
+            for slot in slots {
+                if slot < max_slot {
+                    pubkeys.extend(uncleaned_keys.remove(&slot).unwrap());
+                }
+            }
+        }
+        let delta_key_count = pubkeys.len();
+        let zero_lamport_count = {
+            let zero_lamport_accounts = self.zero_lamport_accounts.read().unwrap();
+            pubkeys.extend(zero_lamport_accounts.iter());
+            zero_lamport_accounts.len()
+        };
+        pubkeys.sort_unstable();
```

Comment:
Could use a `BTreeSet<Slot>` here to avoid the sort and dedup

Reply:
I wrote it with BTreeSet, it's slower overall and spending time constructing the btree inside the locks creates more contetion on those locks.

vec is nice that constructing the unsorted one is very cheap and sort/dedup are pretty cheap and done outside the lock.
```
vec: 23ms sort + 8ms clean + 1.5ms delta = 32ms total
scan: 227ms

btree: 0ms sort + 60ms clean + 11ms delta = 71ms total
scan: 228ms
```
Scan is a similar speed in both cases.